src/displayapp/screens/Music.cpp

```diff
@@ -150,74 +158,94 @@ Music::~Music() {
 }
 
 void Music::Refresh() {
-  if (artist != musicService.getArtist()) {
+  bleState = bleController.IsConnected();
+  if (bleState.Get() == false) {
+    SetDisconnectedUI();
+    lastConnected = false;
+  } else {
+    if (!lastConnected) {
+      // just reconnected
+      musicService.event(Controllers::MusicService::EVENT_MUSIC_OPEN);
+      SetConnectedUI();
+      RefreshTrackInfo(true);
+    } else {
+      RefreshTrackInfo(false);
+    }
+    lastConnected = true;
+  }
+}
+
+void Music::SetDisconnectedUI() {
```

Comment:
If the state variables (playing, artist, track, album etc) were updated here, could we get rid of force? I think the UI will be more robust if the state variables are always kept up to date, that way connection/disconnection don't need special handling

Would also need handling during in the constructor